Microsoft.Toolkit.Uwp.UI.Animations/Xaml/ImplicitAnimationSet.cs

```diff
@@ -6,18 +6,81 @@
 
 using System;
 using System.Diagnostics.Contracts;
+using Windows.Foundation.Collections;
 using Windows.UI.Composition;
 using Windows.UI.Xaml;
 using Windows.UI.Xaml.Hosting;
 
 namespace Microsoft.Toolkit.Uwp.UI.Animations
 {
     /// <summary>
-    /// A collection of implicit animations that can be grouped together. This type represents a composite animation
-    /// (such as <see cref="Windows.UI.Xaml.Media.Animation.Storyboard"/>) that is executed on a given element.
+    /// A collection of implicit animations that can be assigned to a <see cref="UIElement"/> and configured to be run automatically
+    /// when the element is either shown or hidden (through <see cref="Implicit.ShowAnimationsProperty"/> and <see cref="Implicit.HideAnimationsProperty"/>,
+    /// or whenever one of the targeted properties on the underlying <see cref="Visual"/> element changes, through <see cref="Implicit.AnimationsProperty"/>.
+    /// <para>
+    /// Animations within an <see cref="ImplicitAnimationSet"/> should be objects implementing the <see cref="IImplicitTimeline"/> interface, such as
+    /// types inheriting from <see cref="ImplicitAnimation{TValue, TKeyFrame}"/> (eg. <see cref="OpacityAnimation"/>, <see cref="TranslationAnimation"/>,
+    /// <see cref="OffsetAnimation"/> and <see cref="ScaleAnimation"/>, or custom ones such as <see cref="ScalarAnimation"/> and <see cref="Vector3Animation"/>).
+    /// Adding incompatible elements cannot be validated at build-time, but will result in a runtime crash.
+    /// </para>
+    /// <para>
+    /// Animations will monitor for changes in real-time to any of their public properties. For instance, if a binding is used to dynamically update the
+    /// <see cref="Animation{TValue, TKeyFrame}.To"/> to <see cref="Animation{TValue, TKeyFrame}.From"/> properties, the entire animation set will be
+    /// initialized again and assigned to the underlyign <see cref="Visual"/> object for the targeted <see cref="UIElement"/>. This does not currently apply
+    /// to changes to the <see cref="Animation{TValue, TKeyFrame}.KeyFrames"/> property though (other than the entire property being reassigned). To achieve
+    /// dynamic updates to animation sets in that case, either leverage expression keyframes or just use code-behind to manually reinitialize the animations.
+    /// </para>
     /// </summary>
+    /// <remarks>
+    /// An <see cref="ImplicitAnimationSet"/> instance can only be used on a single target <see cref="UIElement"/>, and it cannot be shared across multiple
```

Comment:
Hmm... Is this only if Binding is used or for everything now? As in 6.1.1 (and like we enabled for the Explicit ones) you could define these as resources and apply...

Reply:
Nope, it wasn't supported in 7.0 either. The reason is that `AnimationSet` can either contain a parent reference (if you assing one directly to an item), or no parent references at all (if it's in a resource dictionary), but in that case in order to start the animation you need to explicitly pass the target `UIElement` to animate (which our behaviors do automatically). An implicit set instead has to have a parent reference so that it can use it to recreate the animations whenever it needs to be updated (which was already happening if you removed/added an animation from a set). If it was shared in a resource dictionary then it wouldn't be able to do that. To be clear, due to how the code is currently setup, you _can_ technically share an implicit set in a resource dictionary, but then real-time updates would only work for the _last_ element that the set was applied to. In theory if you never make changes to a set in a dictionary, then it should be fine 🤔

Reply:
@Sergio0694 I was talking about 6.1.1 allows for Implicit animations to be resourced, but you would never have bindings in those scenarios, they're just static.

So I think you're saying it should still allow for that and be fine?

Reply:
Yup, should still work just fine 🙂

Made some changes in 710510117087e6b07bb671db7848c76ab7367776 to make this clearer - basically sharing an implicit set will work, but only the last element the set was attached to will actually receive real-time changes. As in, if you share a set and then use a binding to modify the animations, all elements except the last one the animation set was attached to will just keep using the initial values of the animations. If you don't use bindings instead but just hardcode the animations as before, then you wouldn't notice anything weird going on.